File: java/src/main/java/com/example/java/activity/test/TestContract.java
package com.example.java.activity.test;

import com.android.common.base.BasePresenter;
import com.android.common.base.BaseView;
import com.example.java.bean.ShareData;

import java.util.List;

/**
 * @作者 陈忠岳
 * @主要功能
 * @创建日期 6/2/22
 */

interface TestContract {
    interface View extends BaseView<Presenter> {
        void setShares(List<ShareData> list,Boolean history);
    }
    interface Presenter extends BasePresenter {
        void getShares(String search ,int type);
    }

}

File: kotlin/src/main/java/com/example/kotlin/activity/test/TestContract.kt
package com.example.kotlin.activity.test

import com.android.common.base.BasePresenter
import com.android.common.base.BaseView
import com.example.kotlin.bean.ShareData

/**
 * @作者 陈忠岳
 * @主要功能
 * @创建日期  2019-11-21
 */
interface TestContract {

    interface View : BaseView<Presenter> {
        fun setShares(list: List<ShareData>, history: Boolean = false)
    }

    interface Presenter : BasePresenter {
        fun getShares(search: String, type: Int)
    }

}